lib/widgets/lightbox.dart

```diff
@@ -0,0 +1,243 @@
+import 'content.dart';
+import 'page.dart';
+import 'clipboard.dart';
+import 'store.dart';
+
+// TODO: Add index of the image preview in the message, to not break if
+//   there are multiple image previews with the same URL in the same
+//   message. Maybe keep `src`, so that on exit the lightbox image doesn't
+//   fly to an image preview with a different URL, following a message edit
+//   while the lightbox was open.
+class _LightboxHeroTag {
+  _LightboxHeroTag({required this.messageId, required this.src});
+
+  final int messageId;
+  final String src;
+
+  @override
+  bool operator ==(Object other) {
+    return other is _LightboxHeroTag &&
+      other.messageId == messageId &&
+      other.src == other.src;
+  }
+
+  @override
+  int get hashCode => Object.hash('_LightboxHeroTag', messageId, src);
+}
+
+/// Builds a [Hero] from an image in the message list to the lightbox page.
+class LightboxHero extends StatelessWidget {
+  const LightboxHero({
+    super.key,
+    required this.accountId,
+    required this.message,
+    required this.src,
+  });
+
+  /// The account the message list belongs to.
+  final int accountId;
+
+  final Message message;
+  final String src;
+
+  @override
+  Widget build(BuildContext context) {
+    return Hero(
+      tag: _LightboxHeroTag(messageId: message.id, src: src),
+
+      // RealmContentNetworkImage needs an ancestor PerAccountStoreWidget, and
+      // during the hero's "flight", we wouldn't otherwise get one.
+      child: PerAccountStoreWidget(accountId: accountId,
+        child: RealmContentNetworkImage(src, filterQuality: FilterQuality.medium)),
+    );
+  }
+}
+
+class _CopyLinkButton extends StatelessWidget {
+  const _CopyLinkButton({required this.url});
+
+  final String url;
+
+  @override
+  Widget build(BuildContext context) {
+    return IconButton(
+      tooltip: 'Copy link',
+      icon: const Icon(Icons.copy),
+      onPressed: () async {
+        copyWithPopup(context: context, successText: const Text('Link copied'),
+          data: ClipboardData(text: url));
+      });
+  }
+}
+
+class _LightboxPage extends StatefulWidget {
+  const _LightboxPage({
+    required this.routeEntranceAnimation,
+    required this.message,
+    required this.src,
+  });
+
+  final Animation routeEntranceAnimation;
+  final Message message;
+  final String src;
+
+  @override
+  State<_LightboxPage> createState() => _LightboxPageState();
+}
+
+class _LightboxPageState extends State<_LightboxPage> {
+  // TODO: Animate entrance/exit of header and footer
+  bool _headerFooterVisible = false;
+
+  void _handleRouteEntranceAnimationStatusChange(AnimationStatus status) {
+    switch (status) {
+      case AnimationStatus.completed:
+        // Route entrance transition complete
+        setState(() {
+          _headerFooterVisible = true;
+        });
+        break;
+      default:
+        setState(() {
+          _headerFooterVisible = false;
+        });
+        break;
+    }
```

Comment:
can simplify:

```suggestion
    final entranceAnimationComplete = status == AnimationStatus.completed;
    setState(() {
      _headerFooterVisible = entranceAnimationComplete;
    });
```

(The switch already isn't adding any kind of robustness compared to that, because it has a default case.)